src/test/editor-service/editor-server-test.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from 'chai';
+import * as child_process from 'child_process';
+import * as path from 'path';
+import * as split from 'split';
+
+import {invertPromise} from '../test-utils';
+
+const pathToServer =
+    path.join(__dirname, '../../editor-service/editor-server.js');
+
+suite('RemoteEditorService', function() {
+  /**
+   * These are the tests. We run these tests using a few different ways of
+   * communicating with the server.
+   */
+  function editorServiceInterfaceTests(
+      sendRequest: (request: any) => Promise<void>,
+      getNextResponse: (expectedId: number) => Promise<any>) {
+    const initMessage = {
+      kind: 'init',
+      basedir: `${path.join(__dirname, 'static')}`
+    };
+    const getWarningsMessage = {
+      kind: 'getWarningsFor',
+      localPath: 'malformed.html'
+    };
+
+
+    test('can create and initialize', async function() {
```

Comment:
Here and elsewhere. Done. (I think I did it this way because of a clang-format issue that is now resolved. At least, that's my story and I'm sticking to it.)